javascripts/create.js

```diff
@@ -65,56 +76,67 @@ var Gauntlet = (function(originalGauntlet){
 
 // ----- Function to create new weapon object and add it to character ----- //
 
-  originalGauntlet.chooseWeapon = function(weaponId) {
+  originalGauntlet.chooseWeapon = function() {
     // ---------- Takes the user's choice of weapon and sets up   ---------- //
     // ---------- structure for creating a new weapon that will   ---------- //
     // ---------- be used in the function above to set the weapon ---------- //
+    var y;
 
     switch (weaponId) {
       case "Weapon":
-        var y = new Gauntlet.Armory.Weapon();
+        y = new Gauntlet.Armory.Weapon();
+        break;
+      case "Dagger":
+        y = new Gauntlet.Armory.Dagger();
         break;
       case "BroadSword":
-        var y = new Gauntlet.Armory.BroadSword();
+        y = new Gauntlet.Armory.BroadSword();
         break;
       case "WarAxe":
-        var y = new Gauntlet.Armory.WarAxe();
+        y = new Gauntlet.Armory.WarAxe();
```

Comment:
Notice how the string in each of these case statements is the exact same string of characters as the instance you are creating? Even the case matches. If you think about it, that makes this entire switch statement obsolete and unnecessary.

**Hint:** Square bracket notation for object properties.